client/src/components/AuthProvider.tsx

```diff
@@ -0,0 +1,148 @@
+import { createContext, ReactNode, useContext, useEffect, useState } from 'react';
+import { Alert, CircularProgress, Stack } from '@mui/material';
+import { INFOBIP_API_BASE_URL, CLIENT_ID, REDIRECT_URL } from '../const';
+
+export interface AuthContext {
+  token: string;
+  token_type: string;
+  expires_in: number;
+  accountKey: string;
+  userKey: string;
+  username: string;
+  email: string;
+  locale: string;
+}
+
+const authContext = createContext<AuthContext | null>(null);
+
+export function useAuthContext(): AuthContext {
+  const context = useContext(authContext);
+  if (!context) {
+    throw new Error('useAuthContext can be used only within AuthProvider');
+  }
+
+  return context;
+}
+
+export function AuthProvider({ children }: { children: ReactNode }) {
+  const [authContextState, setAuthContextState] = useState<AuthContext | null>(null);
+  const [isLoading, setLoading] = useState(false);
+  const [error, setError] = useState<any | null>(null);
```

Comment:
Looks like all we are ever setting to `error` (besides `null`) are strings so the type could be `string | null`